pkg/gohai/platform/platform_linux.go

```diff
@@ -5,16 +5,81 @@
 
 package platform
 
-import "strings"
-
-var unameOptions = []string{"-s", "-n", "-r", "-m", "-p", "-i", "-o"}
-
-func updateArchInfo(archInfo map[string]string, values []string) {
-	archInfo["kernel_name"] = values[0]
-	archInfo["hostname"] = values[1]
-	archInfo["kernel_release"] = values[2]
-	archInfo["machine"] = values[3]
-	archInfo["processor"] = values[4]
-	archInfo["hardware_platform"] = values[5]
-	archInfo["os"] = strings.Trim(values[6], "\n")
+import (
+	"bufio"
+	"io"
+	"os"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/gohai/utils"
+	"golang.org/x/sys/unix"
+)
+
+// getUnameOS is similar to `uname -o`
+//
+// uname uses preprocessor macros to determine the printable name of the operating system,
+// cf https://github.com/coreutils/gnulib/blob/master/m4/host-os.m4
+//
+// on every Linux device it prints "GNU/Linux", probably because uname is from GNU coreutils,
+// so if you are using it your OS is considered to be GNU
+func getUnameOS() string {
+	// eventually we might want to return different values depending on the actual OS
+	// (not all Linux are GNU)
+	return "GNU/Linux"
+}
+
+// check if the vendor is AMD
+// reader should be an io.Reader over /proc/cpuinfo
+func isVendorAMD(reader io.Reader) bool {
+	scanner := bufio.NewScanner(reader)
+	for scanner.Scan() {
+		text := scanner.Text()
+		if strings.HasPrefix(text, "vendor_id") {
+			if strings.Contains(text, "AuthenticAMD") {
+				return true
+			}
+			break
+		}
```

Comment:
Could there be a situation where we have a false positive here ? Something like `vendor_id_something    : NonAuthenticAMD` ?

Should we use a regexp to match the entire line instead ?

Reply:
OK. Let's keep it as is then. Just explain that we mimic the behavior of uname in the comment of the function then,

Reply:
Hum too late I pushed a commit doing it properly earlier.
Do you think I should revert to the former implementation ?

I think this one is just more reliable so it seems like an improvement...

Reply:
:thinking: I'm not sure, I think for now we might want to keep the old implementation. We won't know if this is intentional or not. WDYT ?

Reply:
Honestly I think they used `strstr` (which is equivalent to `strings.Contains`) because the function existed so it was easier than splitting on ':', trimming and comparing